V1/Services/OrganisationService.cs

```diff
@@ -114,18 +134,34 @@ public async Task<Response> UpdateCountAsync(string id, UpdateCountRequest paylo
             await _organisationAccessLogRepository.AddAsync(newCount);
 
             await _organisationAccessLogRepository.SaveAsync();
+        }
 
-            var logs = organisation.AccessLogs
-                .Where(oal => oal.CreatedAt.Date.Equals(DateTime.UtcNow.Date))
-                .ToList();
+        private async Task ValidateScan(List<OrganisationAccessLog> logs, ScanType scanType, Wallet wallet, bool mobile = false)
+        {
+            if (wallet != default)
+            {
+                var userLogs = logs
+                    .Where(l => l.Wallet == wallet && l.CreatedAt.Date == DateTime.Now.Date)
+                    .OrderByDescending(l => l.CreatedAt)
+                    .ToList();
 
-            return new Response(
-                new UpdateCountResponse()
-                {
-                    Balance = logs.Count == 0 ? 0 : GetAccessLogBalance(logs)
-                },
-                true,
-                HttpStatusCode.OK);
+                if(!userLogs.Any() && scanType == ScanType.CheckOut && !mobile)
```

Comment:
I would move the !mobile check on its own to the top of the method. It will avoid mobile running through all the checks for mobile

Reply:
True